Add `QNode.update` method (#6803)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Improvements 🛠</h3>

* QNodes now have an :meth:`~.pennylane.QNode.update` method that allows for reconfiguring QNode settings like `diff_method`, `mcm_method`, and more. This allows for easier on-the-fly adjustments to workflows. Any arguments not specified will retain their original value.
  [(#6803)](https://github.com/PennyLaneAI/pennylane/pull/6803)

  After constructing a `QNode`,
  ```python
  import pennylane as qml

  @qml.qnode(device=qml.device("default.qubit"))
  def circuit():
    qml.H(0)
    qml.CNOT([0,1])
    return qml.probs()
  ```

>   its configuration can be changed by using the `QNode.update` method,

Author: ```suggestion
  the configuration can be modified by using the `QNode.update` method,
```

  ```pycon
  >>> print(circuit.diff_method)
  best

>   >>> new_circuit = circuit.update(diff_method="parameter-shift")

Author: ```suggestion
  >>> new_circuit = circuit.update(diff_method="parameter-shift")
  >>> print(new_circuit.diff_method)
```

<h3>Improvements 🛠</h3>

* QNodes now have an :meth:`~.pennylane.QNode.update` method that allows for reconfiguring QNode settings like `diff_method`, `mcm_method`, and more. This allows for easier on-the-fly adjustments to workflows. Any arguments not specified will retain their original value.
  [(#6803)](https://github.com/PennyLaneAI/pennylane/pull/6803)

  After constructing a `QNode`,
  ```python
  import pennylane as qml

  @qml.qnode(device=qml.device("default.qubit"))
  def circuit():
    qml.H(0)
    qml.CNOT([0,1])
    return qml.probs()
  ```

>   the configuration can be modified by using the `QNode.update` method,

Reviewer: ```suggestion
  its settings can be modified with `update` method, which returns a new QNode object. Here is an example
  of updating a QNode's `diff_method`:
```

<h3>Improvements 🛠</h3>

> * `QNode` objects now have an `update` method that allows for re-configuring settings like `diff_method`, `mcm_method`, and more. This allows for easier on-the-fly adjustments to workflows. Any arguments not specified will retain their original value.

Reviewer: ```suggestion
* `QNode` objects now have an :meth:`~.QNode.update` method that allows for re-configuring settings like `diff_method`, `mcm_method`, and more. This allows for easier on-the-fly adjustments to workflows. Any arguments not specified will retain their original value.
```

Reviewer: The sphinx directive, you mean?

Author: Yep!

Reviewer 2: the changelog is markdown, not rst.

> * QNodes now have an :meth:`~.pennylane.QNode.update` method that allows for reconfiguring QNode settings like `diff_method`, `mcm_method`, and more. This allows for easier on-the-fly adjustments to workflows. Any arguments not specified will retain their original value.

Author: ```suggestion
* QNodes now have an `update` method that allows for re-configuring settings like `diff_method`, `mcm_method`, and more. This allows for easier on-the-fly adjustments to workflows. Any arguments not specified will retain their original value.
```

<h3>Improvements 🛠</h3>

> * Add `QNode.update()`, which allows the creation of a new `QNode` instance with modified constructor arguments.

Reviewer 3: One last thing I forgot to suggest.
```suggestion
* Added `QNode.update()`, which allows the creation of a new `QNode` instance with modified constructor arguments.
```

> * Added `QNode.update()`, which allows the creation of a new `QNode` instance with modified constructor arguments.

Author: ```suggestion
* Added `QNode.update(**kwargs)`, which allows the creation of a new `QNode` instance with modified constructor arguments. Any arguments not specified will retain their original value.
```

> * Add `QNode.update` as an instance method for `QNode`, allowing the creation of a new instance with modified constructor arguments.

Reviewer 3: ```suggestion
* Add `QNode.update()`, which allows the creation of a new `QNode` instance with modified constructor arguments.
```

Some of the wording seems a bit redundant

> * Added `QNode.update(**kwargs)`, which allows the creation of a new `QNode` instance with modified constructor arguments. Any arguments not specified will retain their original value.

Reviewer: ```suggestion
* QNodes now have an :meth:`~.pennylane.QNode.update` method that allows for reconfiguring QNode settings like `diff_method`, `mcm_method`, and more. This allows for easier on-the-fly adjustments to workflows. Any arguments not specified will retain their original value.
```

Reviewer: Adding a code example here would be nice I think!

Author: Good idea, I'll do that :)
